arviz/plots/distplot.py

```diff
@@ -83,73 +84,41 @@ def plot_dist(
     -------
     ax : matplotlib axes
     """
-    if ax is None:
-        ax = plt.gca()
 
-    if hist_kwargs is None:
-        hist_kwargs = {}
-    hist_kwargs.setdefault("bins", None)
-    hist_kwargs.setdefault("cumulative", cumulative)
-    hist_kwargs.setdefault("color", color)
-    hist_kwargs.setdefault("label", label)
-    hist_kwargs.setdefault("rwidth", 0.9)
-    hist_kwargs.setdefault("align", "left")
-    hist_kwargs.setdefault("density", True)
+    dist_plot_args = dict(
+        # Internal API
+        # User Facing API that can be simplified
+        values=values,
+        values2=values2,
+        color=color,
+        kind=kind,
+        cumulative=cumulative,
+        label=label,
+        rotated=rotated,
+        rug=rug,
+        bw=bw,
+        quantiles=quantiles,
+        contour=contour,
+        fill_last=fill_last,
+        textsize=textsize,
+        plot_kwargs=plot_kwargs,
+        fill_kwargs=fill_kwargs,
+        rug_kwargs=rug_kwargs,
+        contour_kwargs=contour_kwargs,
+        hist_kwargs=hist_kwargs,
+        ax=ax,
+    )
 
-    if plot_kwargs is None:
-        plot_kwargs = {}
-
-    if rotated:
-        hist_kwargs.setdefault("orientation", "horizontal")
-    else:
-        hist_kwargs.setdefault("orientation", "vertical")
-
-    if kind == "auto":
-        kind = "hist" if values.dtype.kind == "i" else "density"
-
-    if kind == "hist":
-        _histplot_op(
-            values=values, values2=values2, rotated=rotated, ax=ax, hist_kwargs=hist_kwargs
-        )
-    elif kind == "density":
-        plot_kwargs.setdefault("color", color)
-        legend = label is not None
-
-        plot_kde(
-            values,
-            values2,
-            cumulative=cumulative,
-            rug=rug,
-            label=label,
-            bw=bw,
-            quantiles=quantiles,
-            rotated=rotated,
-            contour=contour,
-            legend=legend,
-            fill_last=fill_last,
-            textsize=textsize,
-            plot_kwargs=plot_kwargs,
-            fill_kwargs=fill_kwargs,
-            rug_kwargs=rug_kwargs,
-            contour_kwargs=contour_kwargs,
-            ax=ax,
-        )
-    return ax
+    if backend is None or backend.lower() in ("mpl", "matplotlib"):
```

Comment:
How about reading `rcParams["plot.backend"]` when it is `None`?

Reply:
Yes, this sound good. Also there are now many things we should define for rcParams